### PR TITLE
Added Downloader pySmartDL

### PR DESCRIPTION
**Pull request by ArjixGamer**

### CODE REVIEW

**Comment by ArjixGamer** [comment]
504 Gateway Timeout

**Comment by ArjixGamer** [comment]
I didn't add it to the anime.py Download class because it would replace the internal downloader, but its just a simple edit

File: anime_downloader/downloader/SmartDL.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            headers['Referer'] = self.source.referer
+
+        url = self.source.stream_url
+        request_args = {'headers': headers}
+
+        dest = str(self.path) # str(path.parent.absolute())
+        obj = SmartDL(url, dest, request_args=request_args, progress_bar=True)
+        obj.start()
+        time.sleep(1)
+        sys.exit(0)
```
Comment (vn-ki) [inline anime_downloader/downloader/SmartDL.py:30]: Why are you even exiting here?

File: anime_downloader/downloader/SmartDL.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        if self.source.referer:
+            headers['Referer'] = self.source.referer
+
+        url = self.source.stream_url
+        request_args = {'headers': headers}
+
+        dest = str(self.path) # str(path.parent.absolute())
+        obj = SmartDL(url, dest, request_args=request_args, progress_bar=True)
+        obj.start()
+        time.sleep(1)
```
Comment (vn-ki) [inline anime_downloader/downloader/SmartDL.py:29]: Why sleep?

File: anime_downloader/downloader/SmartDL.py (hunk 1 above)
Comment (ArjixGamer) [reply anime_downloader/downloader/SmartDL.py:30]: thats what the other files had so i just copied it

File: anime_downloader/downloader/SmartDL.py (hunk 2 above)
Comment (ArjixGamer) [reply anime_downloader/downloader/SmartDL.py]: /shrug

**Comment by ArjixGamer** [comment]
made some changes because as it seems i forgot to rename some variables and that ducked it up

**Comment by yetibinary** [comment]
cool

**Pull request merged**